torch/distributed/pipeline/sync/microbatch.py

```diff
@@ -143,7 +143,7 @@ def _setitem_by_index(self, index: int, value) -> None:
         self._values = value
 
     def _setitem_by_slice(self, index: slice, value) -> None:
-        if not (index.start is index.stop is index.step is None):
+        if not (index.start is index.stop is index.step is None):  # noqa: E714
```

Comment:
May want to file this issue with ruff ;)

Reply:
https://github.com/charliermarsh/ruff/issues/4082